### ISSUE TITLE
Q: How to enable 'GCP credential authentication' in istio-1.0.2 ? 

### ISSUE THREAD

**Issue by kidiyoor** [istio/istio#8958]
**Describe the bug**
node_agent log bellow:

`node-agent --ca-address istio-citadel:8060 --env gcp`

`
2018-09-25T21:02:37.608569Z     info    Sending CSR (retrial # 5) ...2018-09-25T21:02:37.608649Z     error   Node agent terminated with error: node agent can't get the CSR approved from Istio CA after max number of retries (5).# in pid 23265, pgrp=23178 -> 232652018-09-25T21:21:11.692848Z     error   GCP credential authentication in CSR API is disabled# in pid 23398, pgrp=23311 -> 233982018-09-25T21:21:53.520356Z     error   GCP credential authentication in CSR API is disabled
`

**Comment by wattli** [comment]
This is indeed an issue and we are actively working on it now.

**Comment by kidiyoor** [comment]
is there a workaround for this in istio-1.0.2 ? basically - to run node_agent on a GCE VM

**Comment by wattli** [comment]
If you treat GCE as raw VM, and manually bootstrap some key/cert, it would work

**Comment by vilobhmm** [comment]
@wattli : I am able to getting it working for 1.0.0. What exactly has changed with 1.0.2 ?

**Comment by wattli** [comment]
It's disabled here: https://github.com/istio/istio/blob/master/security/pkg/platform/client.go#L47

We will bring it back soon.

**Comment by kidiyoor** [comment]
I set --env=onprem
Copied key and certs from secret: istio.default
I am seeing the following issue - 

```
2018-09-26T01:06:15.626589Z     info    grpc: addrConn.createTransport faile
d to connect to {istio-citadel:8060 0  <nil>}. Err :connection error: desc =
 "transport: authentication handshake failed: x509: certificate is not valid
 for any names, but wanted to match istio-citadel". Reconnecting...
2018-09-26T01:06:15.626683Z     info    pickfirstBalancer: HandleSubConnStat
eChange: 0xc4201f6d50, TRANSIENT_FAILURE
2018-09-26T01:06:17.626643Z     info    Sending CSR (retrial #2) ...
2018-09-26T01:06:17.626711Z     error   CSR signing failed: rpc error: code 
= Unavailable desc = all SubConns are in TransientFailure, latest connection
 error: connection error: desc = "transport: authentication handshake failed
: x509: certificate is not valid for any names, but wanted to match istio-ci
tadel". Will retry in 20s
2018-09-26T01:06:27.943352Z     info    pickfirstBalancer: HandleSubConnStat
eChange: 0xc4201f6d50, CONNECTING
2018-09-26T01:06:28.137558Z     info    grpc: addrConn.createTransport faile
d to connect to {istio-citadel:8060 0  <nil>}. Err :connection error: desc =
 "transport: authentication handshake failed: x509: certificate is not valid
 for any names, but wanted to match istio-citadel". Reconnecting...
2018-09-26T01:06:28.137641Z     info    pickfirstBalancer: HandleSubConnStat
eChange: 0xc4201f6d50, TRANSIENT_FAILURE
```
```
[mtls-vk ~]$ sudo openssl x509 -text -noout -in /etc/certs/root-cert.pem 
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            20:fd:34:b1:b1:2e:6d:4a:85:cb:4c:a4:c9:56:67:f7
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: O=k8s.cluster.local
        Validity
            Not Before: Sep 26 00:08:26 2018 GMT
            Not After : Sep 26 00:08:26 2019 GMT
        Subject: O=k8s.cluster.local
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
```
```
[mtls-vk ~]$ sudo openssl x509 -text -noout -in /etc/certs/cert-chain.pem 
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            f5:0a:47:7f:71:b1:d7:6b:ad:3f:6c:e6:18:a2:55:b8
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: O=k8s.cluster.local
        Validity
            Not Before: Sep 25 16:57:17 2018 GMT
            Not After : Dec 24 16:57:17 2018 GMT
        Subject: O=
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
```

**Comment by wattli** [comment]
https://github.com/istio/istio/issues/8951, this is a regression and @myidpt is on it now.

**Comment by myidpt** [comment]
Sorry that was a bug introduced by disabling the GCP ID token authentication. We have a PR and it will be fixed in 1.0.3.

**Comment by wattli** [comment]
The PR was submitted, please either try the latest daily build https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/ or wait until 1.0.3 release.

**Comment by wattli** [comment]
Currently the on-prem provisioning is done, however, the GCE bug is still not fixed.

**Comment by kidiyoor** [comment]
That means - if we treat GCE VM a regular VM ie set ENV=onprem and manually provision the certs, it should work fine ?

Are we expecting GCE bug to be fixed in 1.0.3 ?

**Comment by kidiyoor** [comment]
Now that we are using ENV='onprem' - We are manually copying the private key from secret to the VM the first time. What happens to the node_agent when the key is rotated in the k8s cluster ?
